File: qc-core/src/main/java/com/eager/questioncloud/core/domain/creator/infrastructure/CreatorRepository.java
package com.eager.questioncloud.core.domain.creator.infrastructure;

import com.eager.questioncloud.core.domain.creator.dto.CreatorInformation;
import com.eager.questioncloud.core.domain.creator.model.Creator;

public interface CreatorRepository {
    Boolean existsById(Long creatorId);

    CreatorInformation getCreatorInformation(Long creatorId);

    Creator save(Creator creator);

    void deleteAllInBatch();
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/creator/infrastructure/repository/CreatorRepository.kt
package com.eager.questioncloud.core.domain.creator.infrastructure.repository

import com.eager.questioncloud.core.domain.creator.dto.CreatorInformation
import com.eager.questioncloud.core.domain.creator.model.Creator

interface CreatorRepository {
    fun existsById(creatorId: Long): Boolean

    fun getCreatorInformation(creatorId: Long): CreatorInformation

    fun save(creator: Creator): Creator

    fun deleteAllInBatch()
}
